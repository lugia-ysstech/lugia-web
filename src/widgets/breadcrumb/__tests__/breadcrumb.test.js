import React from 'react';

import Breadcrumb from '../';
import InnerBreadcrumb from '../breadcrumb';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Icon from '../../icon';

Enzyme.configure({ adapter: new Adapter() });

const routes = [
  {
    path: 'index',
    breadcrumbName: '首页',
  },
  {
    path: 'first',
    breadcrumbName: '一级面包屑',
  },
  {
    path: 'second',
    breadcrumbName: '二级面包屑',
  },
  {
    path: 'third',
    breadcrumbName: '当前页面',
  },
];

describe('Breadcrumb', () => {
  it('<Breadcrumb>和<Breadcrumn.Item>配合使用', () => {
    expect(
      renderer
        .create(
          <Breadcrumb>
            <Breadcrumb.Item>主页</Breadcrumb.Item>
            <Breadcrumb.Item>一级菜单</Breadcrumb.Item>
            <Breadcrumb.Item>二级菜单</Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb> 中separator分隔符', () => {
    expect(
      renderer
        .create(
          <Breadcrumb separator={'>'}>
            <Breadcrumb.Item>主页</Breadcrumb.Item>
            <Breadcrumb.Item>一级菜单</Breadcrumb.Item>
            <Breadcrumb.Item>二级菜单</Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb> 中传入图标', () => {
    expect(
      renderer
        .create(
          <Breadcrumb separator={'>'}>
            <Breadcrumb.Item>
              <Icon iconClass="lugia-icon-reminder_close_circle_o" type="home" />
              <span>主页</span>
            </Breadcrumb.Item>
            <Breadcrumb.Item>
              <span>一级面包屑菜单</span>
              <Icon iconClass="lugia-icon-reminder_close_circle_o" type="home" />
            </Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb.Item> 中传入href属性，文字变成超链接', () => {
    expect(
      renderer
        .create(
          <Breadcrumb separator={'>'}>
            <Breadcrumb.Item href="">主页</Breadcrumb.Item>
            <Breadcrumb.Item href="">一级菜单</Breadcrumb.Item>
            <Breadcrumb.Item href="">二级菜单</Breadcrumb.Item>
          </Breadcrumb>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it('<Breadcrumb> 传入routes参数直接生成 面包屑', () => {
    expect(
      renderer.create(<Breadcrumb separator={'>'} routes={routes} />).toJSON()
    ).toMatchSnapshot();
  });

  it('最后一项也显示分隔符, lastSeparator', () => {
    expect(
      renderer.create(<Breadcrumb separator={'>'} lastSeparator={'>'} routes={routes} />).toJSON()
    ).toMatchSnapshot();
  });

  it('Breadcrumb.prototype.getBreadCrumbItemConfig only href', () => {
    expect(
      InnerBreadcrumb.prototype.getBreadCrumbItemConfig([
        {
          href: 'index',
          title: '首页',
        },
        {
          href: 'first',
          title: '一级面包屑',
        },
        {
          href: 'second:id',
          title: '二级面包屑',
        },
        {
          href: 'third',
          title: '当前页面',
        },
      ])
    ).toEqual([
      {
        title: '首页',
        isLast: false,
        href: '/index',
        icons: {},
      },
      {
        href: '/first',
        isLast: false,
        title: '一级面包屑',
        icons: {},
      },
      {
        isLast: false,
        href: '/second:id',
        title: '二级面包屑',
        icons: {},
      },
      {
        href: '/third',
        title: '当前页面',
        isLast: true,
        icons: {},
      },
    ]);
  });
  it('Breadcrumb.prototype.getBreadCrumbItemConfig only path', () => {
    expect(
      InnerBreadcrumb.prototype.getBreadCrumbItemConfig([
        {
          path: 'index',
          title: '首页',
        },
        {
          path: 'first',
          title: '一级面包屑',
        },
        {
          path: 'second:id',
          title: '二级面包屑',
        },
        {
          path: 'third',
          title: '当前页面',
        },
      ])
    ).toEqual([
      {
        title: '首页',
        isLast: false,
        href: '/index',
        icons: {},
      },
      {
        href: '/index/first',
        isLast: false,
        title: '一级面包屑',
        icons: {},
      },
      {
        isLast: false,
        href: '/index/first/second:id',
        title: '二级面包屑',
        icons: {},
      },
      {
        href: '/index/first/second:id/third',
        title: '当前页面',
        isLast: true,
        icons: {},
      },
    ]);
  });
  it('Breadcrumb.prototype.getBreadCrumbItemConfig path & href', () => {
    expect(
      InnerBreadcrumb.prototype.getBreadCrumbItemConfig([
        {
          path: 'index',
          title: '首页',
        },
        {
          path: 'first',
          title: '一级面包屑',
        },
        {
          path: 'second:id',
          title: '二级面包屑',
          href: 'hello',
        },
        {
          path: 'third',
          title: '当前页面',
        },
      ])
    ).toEqual([
      {
        title: '首页',
        href: '/index',
        isLast: false,
        icons: {},
      },
      {
        href: '/index/first',
        title: '一级面包屑',
        isLast: false,
        icons: {},
      },
      {
        href: '/hello',
        title: '二级面包屑',
        isLast: false,
        icons: {},
      },
      {
        href: '/index/first/second:id/third',
        title: '当前页面',
        isLast: true,
        icons: {},
      },
    ]);
  });

  it('Breadcrumb.prototype.getBreadCrumbItemConfig only href', () => {
    expect(
      InnerBreadcrumb.prototype.getBreadCrumbItemConfig(
        [
          {
            href: 'index/:name',
            title: '首页:name',
          },
          {
            href: 'first',
            title: '一级面包屑',
          },
          {
            href: 'second/:id',
            title: '二级面包屑',
          },
          {
            href: 'third',
            title: '当前页面',
          },
        ],
        { id: '1', name: 'ligx' }
      )
    ).toEqual([
      {
        title: '首页ligx',
        isLast: false,
        href: '/index/ligx',
        icons: {},
      },
      {
        href: '/first',
        isLast: false,
        title: '一级面包屑',
        icons: {},
      },
      {
        isLast: false,
        href: '/second/1',
        title: '二级面包屑',
        icons: {},
      },
      {
        href: '/third',
        title: '当前页面',
        isLast: true,
        icons: {},
      },
    ]);
  });

  it('Breadcrumb.prototype.getBreadCrumbItemConfig no path & no href', () => {
    expect(
      InnerBreadcrumb.prototype.getBreadCrumbItemConfig([
        {
          path: 'index',
          title: '首页',
        },
        {
          title: '一级面包屑',
        },
        {
          path: 'second:id',
          title: '二级面包屑',
          href: 'hello',
        },
        {
          path: 'third',
          title: '当前页面',
        },
      ])
    ).toEqual([
      {
        title: '首页',
        href: '/index',
        isLast: false,
        icons: {},
      },
      {
        href: undefined,
        title: '一级面包屑',
        isLast: false,
        icons: {},
      },
      {
        href: '/hello',
        title: '二级面包屑',
        isLast: false,
        icons: {},
      },
      {
        href: '/index/second:id/third',
        title: '当前页面',
        isLast: true,
        icons: {},
      },
    ]);
  });
});
