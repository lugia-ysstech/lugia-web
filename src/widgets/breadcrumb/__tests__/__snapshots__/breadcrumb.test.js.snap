// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb <Breadcrumb.Item> 中传入href属性，文字变成超链接 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: font-size 0.3s;
  transition: font-size 0.3s;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 1rem;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  height: 3rem;
  display: inline-block;
  box-sizing: border-box;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  width: 100%;
  line-height: 3rem;
}

.c2 {
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          主页
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          一级菜单
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#000",
            }
          }
        >
          二级菜单
        </a>
        <span
          className="c6"
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Breadcrumb <Breadcrumb> 中separator分隔符 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: font-size 0.3s;
  transition: font-size 0.3s;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 1rem;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  height: 3rem;
  display: inline-block;
  box-sizing: border-box;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  width: 100%;
  line-height: 3rem;
}

.c2 {
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          主页
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          一级菜单
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#000",
            }
          }
        >
          二级菜单
        </a>
        <span
          className="c6"
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Breadcrumb <Breadcrumb> 中传入图标 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: font-size 0.3s;
  transition: font-size 0.3s;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 1rem;
  margin-right: 1rem;
}

.c7 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  height: 3rem;
  display: inline-block;
  box-sizing: border-box;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  width: 100%;
  line-height: 3rem;
}

.c2 {
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
}

.c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          <i
            className="lugia-icon-reminder_close_circle_o   c5"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={Object {}}
          />
          <span>
            主页
          </span>
        </a>
        <span
          className="c6"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#000",
            }
          }
        >
          <span>
            一级面包屑菜单
          </span>
          <i
            className="lugia-icon-reminder_close_circle_o   c5"
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            style={Object {}}
          />
        </a>
        <span
          className="c7"
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Breadcrumb <Breadcrumb> 传入routes参数直接生成 面包屑 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: font-size 0.3s;
  transition: font-size 0.3s;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 1rem;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  height: 3rem;
  display: inline-block;
  box-sizing: border-box;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  width: 100%;
  line-height: 3rem;
}

.c2 {
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index/first"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index/first/second"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index/first/second/third"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </a>
        <span
          className="c6"
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Breadcrumb <Breadcrumb>和<Breadcrumn.Item>配合使用 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: font-size 0.3s;
  transition: font-size 0.3s;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 1rem;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  height: 3rem;
  display: inline-block;
  box-sizing: border-box;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  width: 100%;
  line-height: 3rem;
}

.c2 {
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          主页
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          /
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          一级菜单
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          /
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#000",
            }
          }
        >
          二级菜单
        </a>
        <span
          className="c6"
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Breadcrumb 最后一项也显示分隔符, lastSeparator 1`] = `
.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: font-size 0.3s;
  transition: font-size 0.3s;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-size: 1.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 1rem;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.6rem;
  display: inline-block;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  height: 3rem;
  display: inline-block;
  box-sizing: border-box;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
  width: 100%;
  line-height: 3rem;
}

.c2 {
  display: inline-block;
  padding: 0;
  box-sizing: border-box;
  vertical-align: top;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  box-sizing: border-box;
}

<div
  className="c0"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index/first"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index/first/second"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          
        </a>
        <span
          className="c5"
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
    <div
      className="c2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3"
      >
        <a
          className="c4"
          href="/index/first/second/third"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#000",
            }
          }
        >
          
        </a>
        <span
          className="c6"
          style={
            Object {
              "color": "#000",
            }
          }
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
</div>
`;
